File: JavaScript/Data Structures/Linear/Queue/Linear Queue.js
class LinearQueue {
    constructor() {
        this.queue = [];
        this.front = -1;
        this.rear = -1;
    }

    IsEmpty() {
        return this.rear < 0;
    }

    Enqueue(value) {
        this.queue.push(value);
        this.rear += 1;
    }

    Dequeue() {
        if(this.IsEmpty()) {
            console.log('Queue is Empty');
        } else {
            let temp = this.queue[0];
            for(let i = 0; i < this.rear; i++) {
                this.queue[i] = this.queue[i+1];
            }
            this.rear -= 1;
            this.queue.pop();
            return temp;
        }
    }

    GetFront() {
        if(this.IsEmpty()) {
            return -1;
        } else {
            return this.front+1;
        }
    }

    GetRear() {
        if(this.IsEmpty()) {
            return -1;
        } else {
            return this.rear;
        }
    }

    Clear() {
        this.queue = [];
        this.front = -1;
        this.rear = -1;
    }

    Count() {
        return this.rear+1;
    }

    DisplayQueue() {
        console.log(this.queue);
    }
}


// Testing.
let queue = new LinearQueue();
console.log(queue.Count());
queue.Enqueue(10);
queue.Enqueue(14);
queue.Enqueue(36);
queue.Enqueue(-10);
console.log(queue.Count());
queue.DisplayQueue();
console.log(queue.Dequeue());
console.log(queue.Dequeue());
console.log(queue.GetFront());
console.log(queue.GetRear());
console.log(queue.Count());
queue.Enqueue(40);
console.log(queue.Count());
queue.DisplayQueue();
console.log(queue.GetFront());
console.log(queue.GetRear());
queue.Clear();
console.log(queue.Count());

File: C#/Data Structures/Linear/Queue/Linear Queue.cs
using System;

namespace QueueUsingArray
{
    class Program
    {
        public static void Main(string[] args)
        {
            LinearQueue queue = new LinearQueue(10);
            Console.WriteLine("Size of Queue: " + queue.GetSize());
            queue.Enqueue(10);
            queue.Enqueue(20);
            queue.Enqueue(-30);
            queue.Enqueue(5);
            Console.WriteLine("Size of Queue: " + queue.GetSize());
            queue.GetRear();
            queue.GetFront();
            queue.Dequeue();
            queue.Dequeue();
            Console.WriteLine("Size of Queue: " + queue.GetSize());
            queue.GetRear();
            queue.GetFront();
            queue.Clear();
            Console.WriteLine("Size of Queue: " + queue.GetSize());
        }
    }

    class LinearQueue
    {
        private int front;
        private int rear;
        private int capacity;
        private int[] queue;

        public LinearQueue(int capacity)
        {
            this.front = this.rear = -1;
            this.capacity = capacity;
            this.queue = new int[capacity];
        }

        public Boolean IsEmpty()
        {
            return this.rear < 0;
        }

        public Boolean IsFull()
        {
            return this.rear >= this.capacity-1;
        }

        public void Enqueue(int value)
        {
            if (this.IsFull())
            {
                Console.WriteLine("Queue is Full");
            }
            else
            {
                queue[++this.rear] = value;
            }
        }

        public void Dequeue()
        {
            if (this.IsEmpty())
            {
                Console.WriteLine("Queue is Empty");
            }
            else
            {
                for (int i = 0; i <= this.rear; i++)
                {
                    queue[i] = queue[i + 1];
                }
                queue[rear --] = 0;
            }
        }

        public void GetFront()
        {
            if (IsEmpty())
            {
                Console.WriteLine("Queue is Empty");
            }
            else
            {
                Console.WriteLine($"Queue front is {queue[0]}");
            }
        }

        public void GetRear()
        {
            Console.WriteLine($"Queue rear is {queue[rear]}");
        }

        public int GetSize()
        {
            return this.rear + 1;
        }

        public void Clear()
        {
            this.rear = this.front = -1;
            Array.Clear(this.queue, 0, this.capacity);
        }
    }
}